exasol_advanced_analytics_framework/query_handler/context/top_level_query_handler_context.py

```diff
@@ -29,6 +31,26 @@ def get_current_value(self) -> int:
         return result
 
 
+class ChildContextNotReleasedError(Exception):
+
+    def __init__(self,
+                 not_released_child_contexts: List[ScopeQueryHandlerContext],
+                 exceptions_from_not_released_child_contexts: List["ChildContextNotReleasedError"]):
+        self.exceptions_from_not_released_child_contexts = exceptions_from_not_released_child_contexts
+        self.not_released_child_contexts = not_released_child_contexts
+        concatenated_contexts = "\n- ".join([str(c) for c in self.get_all_not_released_contexts()])
+        self.message = \
+            f"Following child contexts were not released:\n" \
+            f"- {concatenated_contexts}\n"
+        super(ChildContextNotReleasedError, self).__init__(self.message)
+
+    def get_all_not_released_contexts(self):
```

Comment:
```suggestion
    def get_number_of_not_released_contexts(self):
```
Otherwise the name implies you will get a list.

Reply:
you get a list